tools/base/fpgainfo/tempinfo.c

```diff
@@ -56,28 +56,42 @@ static void print_temp_info(fpga_token token)
 	uint64_t num_metrics_info;
 	fpga_result res = FPGA_OK;
 	uint64_t pkg_temp;
+	struct stat st;
 
 	res = fpgaGetProperties(token, &props);
 	ON_FPGAINFO_ERR_GOTO(res, out_destroy, "Failure reading properties from token");
 
 	fpgainfo_board_info(token);
 	fpgainfo_print_common("//****** TEMP ******//", props);
 
-	res = fpgaTokenGetObject(token, PKG_TEMP_NAME, &obj, FPGA_OBJECT_GLOB);
-	ON_FPGAINFO_ERR_GOTO(res, out_destroy, "Failure getting temp object from token");
-	res = fpgaObjectRead64(obj, &pkg_temp, FPGA_OBJECT_SYNC);
-	ON_FPGAINFO_ERR_GOTO(res, out_destroy, "Failure reading package temperature value");
+	if (!stat("/sys/class/fpga", &st)) {
```

Comment:
Seems like this needs an abstraction.  Why do we need to look in sysfs?

Reply:
Can't we come up with a better way to do this?

Reply:
/sys/bus/pci/drivers/intel-fpga-pci Intel FPGA driver 

/sys/bus/pci/drivers/dfl-pci  DFL Upstream driver

we have to implement common library in tools, it provides API to get type of driver etc.

fixed